addons/website_mass_mailing/static/src/xml/website_mass_mailing.xml

```diff
@@ -1,14 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates id="template" xml:space="preserve">
-<t t-name="website_mass_mailing.edition.wrapper">
-    <div role="dialog" class="modal-dialog modal-md d-block" id="o_newsletter_popup">
-        <div class="modal-content o_popup_modal_content">
-            <div class="o_popup_modal_body text-center">
-                <div class="o_popup_content_dev">
-                    <div id="wrapper"></div>
+
+    <t t-name="website_mass_mailing.popup">
+        <div class="modal fade o_newsletter_modal" t-att-id="modalID" t-attf-data-backdrop="#{isEditMode ? 'false' : 'true'}" role="dialog">
+            <div class="modal-dialog modal-dialog-centered" role="document">
+                <div class="modal-content rounded-0">
```

Comment:
With the `Dialog` class whenever we open dialog it is appended to the body so the content inside the dialog is no longer editable.
Also, we need to do lots of modification in `$modal` element like removing header and footer, set backdrop attribute and few more.
we need to pass `$target` When we initialize the `Dialog` and then we need to override `destroy` method and set updated content to `$target` by `.data()`.
To simplify this I've used the standard bootstrap structure.
Correct me if I'm wrong?

Reply:
You can add new options to the `Dialog` class directly
- Where it should be appended (default to body)
- Whether a header / footer should be display
- If we need a backdrop or not (default to true)
- ...

The code may not be simpler but it would not duplicate the modal structure and it would be reusable for every dialog.  What do you think ?

Reply:
I've found that the bootstrap is appending model only in the body see:
https://github.com/odoo/odoo/blob/master/addons/web/static/lib/bootstrap/js/modal.js#L283
in this commit, I've added option via monkey patch to append modal to different target
https://github.com/odoo-dev/odoo/commit/ae708e5ddabb3089f0572d72f83d958c5bbf7e68
can you give your thoughts on this? :|

Reply:
Your link shows that BS is only appending the modal in the body if it is not already in the DOM. I think that if you just add
```javascript
this.$modal.appendTo($parentNode);
```
before using `.modal('show')`, it should work. Is this not the case ?

Reply:
Done